.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,57 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+// README at: https://github.com/devcontainers/templates/tree/main/src/python
+{
+	"name": "Python 3",
+	// Or use a Dockerfile or Docker Compose file. More info: https://containers.dev/guide/dockerfile
+	"image": "mcr.microsoft.com/devcontainers/python:1-3.11-bullseye",
+
+	// Features to add to the dev container. More info: https://containers.dev/features.
+	"features": {
+		"ghcr.io/devcontainers/features/node:1": {}
+	},
+
+	"forwardPorts": [4141, 4142],
+
+	"portsAttributes": {
+		"4142": {
+			"label": "Backend",
+			"onAutoForward": "notify"
+		},
+		"4141": {
+			"label": "Frontend",
+			"onAutoForward": "openBrowser"
+		}
+	},
+
+	"containerEnv": {
+		"DISPLAY": "dummy",
+		"PYTHONUNBUFFERED": "True",
+		"UV_LINK_MODE": "copy",
+		"VIRTUAL_ENV": "/home/vscode/venv"
+	},
+
+	// FIXME: Why does it say "Property build is not allowed" despite it being part of the standard and the docs?
+	// https://containers.dev/implementors/json_reference/#image-specific
+	// https://code.visualstudio.com/docs/devcontainers/create-dev-container#_dockerfile
+	"build": {
+		"args": {
+			"NODE_VERSION": "18"
+		}
+	},
+
+	"onCreateCommand": "./.devcontainer/onCreateCommand.sh",
+
+	"postStartCommand": {
+		"backend": "nohup bash -c 'uv run python package/kedro_viz/server.py demo-project &' > backend.out 2> backend.err < /dev/null",
```

Comment:
I am a little confused why this happens, from where does it get the build files?

Reply:
I think it's getting them from the published version 😓